src/role_carry.js

```diff
@@ -163,6 +184,27 @@ roles.carry.preMove = function(creep, directions) {
   }
 
   reverse = creep.pickupWhileMoving(reverse);
+
+  // recycle carry
+  if (creep.memory.recycle) {
+    reverse = true;
+    creep.say('recycle');
+    if (creep.inBase()) {
+      if (creep.ticksToLive > config.carry.recycleThreshold && roles.carry.reset(creep)) {
+        creep.memory.recycle = false;
+      } else {
+        const spawn = creep.pos.findClosestByRangePropertyFilter(FIND_MY_STRUCTURES, 'structureType', [STRUCTURE_SPAWN]);
+        if (spawn && spawn.recycleCreep(creep) === 'OK') {
+          return true;
+        }
+      }
+      if (roles.carry.checkOnPathStart(creep)) {
+        creep.memory.moveToSpawn = true;
```

Comment:
it's better to handle carries pre-recycle logic with another memory flag, and set recycle flag here

Reply:
you're right, I'll use `resetTarget` for pre-recycle logic